lib/arsi.rb

```diff
@@ -54,6 +56,13 @@ def enable(&block)
 
     private
 
+    def arel_where_sql(arel, relation)
+      return arel.where_sql if relation.nil?
+      return arel.where_sql unless relation.model.respond_to?(:arel_engine)
+
+      arel.where_sql(relation.model.arel_engine)
```

Comment:
are there models withput arel_engine or is this just preventing nils ? ... then we can simplify to:
```suggestion
      arel.where_sql(relation&.model&.arel_engine)
```

Reply:
As I recall it, it’s about whether the `relation` is _actually_ a `Relation` instance (on which we can call `#arel_engine`), or whether it’s maybe just an `ActiveRecord::Base` subclass.

Reply:
One of the existing test cases caused an exception when the line 61 return isn’t there, so it should be possible to investigate further.